rskj-core/src/main/java/co/rsk/logfilter/BlocksBloom.java

```diff
@@ -71,11 +86,11 @@ public void addBlockBloom(long blockNumber, Bloom blockBloom) {
             this.fromBlock = blockNumber;
             this.toBlock = blockNumber;
             this.empty = false;
-        }
-        else if (blockNumber == toBlock + 1) {
+        } else if (backwardsAddition && blockNumber == toBlock - 1) {
```

Comment:
shouldn't it be `&& blockNumber == fromBlock - 1` ?

Reply:
do we need this `backwardsAddition ` flag? I guess we could simply support addition op in both directions?

Reply:
> shouldn't it be && blockNumber == fromBlock - 1 ?

not sure if I get your point Vlad, but we are adding blocks backwards in this mode, so to ensure no gaps exist, we need to assert that received block is current one minus one.

Reply:
> do we need this backwardsAddition flag? I guess we could simply support addition op in both directions?

I had sth like that before, having just two add methods, but [this comment](https://github.com/rsksmart/rskj/pull/1985#discussion_r1122760985) from @asoto-iov made me realise it could be problematic (and misleader for next dev using it) to allow adding in both directions.

And since for now we don't have the use case of adding in both directions, IMO it's simpler for the dev to know what direction will be used for the whole BlocksBloom lifecycle.

WDYT? Do you see anything I might be missing?